src/proto/generation.proto

```diff
@@ -97,25 +94,53 @@ message SamplerParameters {
   optional float cfg_scale = 5;
 }
 
+// Unused, but reserved for future use. Adjustments to the latents after
+// initialization.
 message ConditionerParameters {
   optional string vector_adjust_prior = 1;
 }
 
+// Future, unimplemented.
 enum Upscaler {
   UPSCALER_RGB = 0;
   UPSCALER_GFPGAN = 1;
   UPSCALER_ESRGAN = 2;
 }
 
+// When does this schedule definition apply?
 message ScheduleParameters {
   optional float start = 1;     // 0.0 to 1.0
   optional float end = 2;       // 0.0 to 1.0
 }
 
+// Parameters that apply to this block of the schedule.
 message StepParameter {
   float scaled_step = 1;
   optional SamplerParameters sampler = 2;
   optional ScheduleParameters schedule = 3;
+  optional GuidanceParameters guidance = 4;
+}
+
+// What type of guidance to use.
+enum GuidanceType {
+  GUIDANCE_CFG = 0;               // default is CFG guidance
+  GUIDANCE_CFG_CLIP = 1;          // CFG and CLIP guidance
+}
+
+// Presets for CLIP guidance.
+enum GuidancePreset {
+  GUIDANCE_PRESET_NONE = 0;
+  GUIDANCE_PRESET_EFFICIENT = 1;
+  GUIDANCE_PRESET_BALANCED = 2;
+  GUIDANCE_PRESET_QUALITY = 3;
+}
+
+// Parameters that affect the behavior of the guidance, typically used for CLIP.
+// The omission of this field implies the default guidance of CFG.
+message GuidanceParameters {
+  optional GuidanceType type = 1;
+  optional GuidancePreset guidance_preset = 2;
+  optional float guidance_strength = 3;          // 0.0 to 1.0
```

Comment:
The comment doesn't seem to match the default value here

Reply:
The value after the `=` is the field number for the protobuf https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers.

Reply:
🤦 sorry my bad